cloudferrylib/utils/drivers/ssh_chunks.py

```diff
@@ -51,128 +55,129 @@ def transfer(self, data):
         ssh_user_dst = self.cfg.dst.ssh_user
         ssh_sudo_pass_dst = self.cfg.dst.ssh_sudo_password
 
-        src_temp_dir = os.path.join(self.cfg.src.temp, '')
-        dst_temp_dir = os.path.join(self.cfg.dst.temp, '')
+        runner_src = remote_runner.RemoteRunner(host_src, ssh_user_src)
+        runner_dst = remote_runner.RemoteRunner(host_dst, ssh_user_dst)
+        temp_dir_name = tempfile.mkdtemp(prefix=self.__class__.__name__)
+        src_temp_dir = os.path.join(temp_dir_name, '')
+        dst_temp_dir = os.path.join(temp_dir_name, '')
 
         attempts_count = self.cfg.migrate.retry
         part_size = self.cfg.migrate.ssh_chunk_size
 
-        part_count, part_modulo = self._calculate_parts_count(data)
-
-        with settings(host_string=host_src,
-                      user=ssh_user_src,
-                      password=ssh_sudo_pass_src), utils.forward_agent(
-                env.key_filename):
-            for part in range(part_count):
-                success = 0  # marker of successful transport operation
-                attempt = 0  # number of retry
-
-                # Create chunk
-                if part == range(part_count)[0]:
-                    # First chunk
-                    command = dd_src_command % (path_src,
+        with files.RemoteDir(runner_src, temp_dir_name):
+            with files.RemoteDir(runner_dst, temp_dir_name):
```

Comment:
this would make an attempt to remove `temp_dir_name` twice, should be fine though, no action needed